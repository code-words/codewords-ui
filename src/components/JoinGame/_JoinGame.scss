.JoinGame {
  position: relative;
  left: 5%;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  height: 100%;
  width: 70%;

  div {
    display: flex;
    flex-direction: column;
    width: 60%;
    
    input {
      width: 200px;
    }
  }

}