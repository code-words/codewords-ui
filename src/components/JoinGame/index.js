import React, { Component } from 'react';
import { API_ROOT, HEADERS } from "../../variables";
import { Redirect } from "react-router-dom";

class JoinGame extends Component {
  constructor() {
    super()
    this.state = {
      inviteCode: "",
      name: "",
      redirect: false
    };
  }

  handleSubmit = (e) => {
    e.preventDefault();
    let { name, inviteCode } = this.state;

    let option = {
      headers: HEADERS,
      method: "POST",
      body: JSON.stringify({ name })
    };
    fetch(`${API_ROOT}/v1/games/${inviteCode}/players`, option)
      .then(response => response.json())
      .then(result => { this.props.handleUserInit(result) });

    this.setState({ redirect: true });
  }

  handleChange = (e) => {
    const { name, value } = e.target; 
    this.setState({ [name]: value })
  }

  render() {
    if (this.state.redirect) { return <Redirect to='/lobby' /> };

    return (
      <div className="backdrop">
        <section className="StartScreen">
          <form className="JoinGame" onSubmit={this.handleSubmit}>
            <h2>Friends started a game?</h2>
            <div>
              <label htmlFor="inviteCode">Please enter invite code</label>
              <input
                name="inviteCode"
                value={this.state.inviteCode}
                type="text"
                placeholder="Enter Invite Code"
                onChange={this.handleChange}
              />
            </div>
            <div>
              <label htmlFor="name">Choose your agent name</label>
              <input
                name="name"
                type="text"
                placeholder="Enter Name"
                maxLength="15"
                onChange={this.handleChange}
              />
            </div>
            <input
              className="btn-default start-btn"
              type="submit"
              value="SUBMIT"
              disabled={!this.state.inviteCode || !this.state.name}
            />
          </form>
        </section>
      </div>
    );
  }
}

export default JoinGame;