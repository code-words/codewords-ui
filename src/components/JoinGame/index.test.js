import JoinGame from './index'
import React from 'react'
import {shallow } from 'enzyme'


describe('JoinGame', () => {
  let wrapper = shallow(<JoinGame/>)
  let event = {target:{value: "test", name: 'name'}, preventDefault: jest.fn()}
  it('should matchsnapShot', () => {
    expect(wrapper).toMatchSnapshot()
  });
  it('should call updateState on submit', () => {
    wrapper.find('.JoinGame').simulate('submit', event)
    expect(wrapper.state('redirect')).toEqual(true)
  });
  it('should update state on change', () => {
    wrapper.find('input').simulate('change', event)
    expect(wrapper.state('name')).toEqual('test')

  });

});