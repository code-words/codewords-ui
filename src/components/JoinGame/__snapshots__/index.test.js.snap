// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinGame should match component snapshot 1`] = `ShallowWrapper {}`;

exports[`JoinGame should matchsnapShot 1`] = `ShallowWrapper {}`;
